torch/jit/__init__.py

```diff
@@ -2079,6 +2085,68 @@ def _get_script_class(name):
                            "Did you forget to import it?".format(name))
     return _script_classes[name]
 
+# qualified_name => list[(overload decl, overload default args)]
+_overloaded_fns = {}
+
+# qualified name => list[compiled fns]
+_compiled_overloaded_fns = {}
+
+def overload(func):
+    qual_name = _qualified_name(func)
+    global _overloaded_fns
+    fn_overload_list = _overloaded_fns.get(qual_name)
+    if fn_overload_list is None:
+        fn_overload_list = []
+        _overloaded_fns[qual_name] = fn_overload_list
+    signature = torch.jit.annotations.get_signature(func)
+    if signature is None:
+        raise RuntimeError("Must explicitly add type annotations to overloaded functions: {obj}").format(func)
+    fn_overload_list.append((torch.jit.get_jit_def(func).decl(), get_default_args(func)))
+    return func
+
+def compile_function_with_overload(qual_name, impl_fn, overload_decl, overload_defaults):
```

Comment:
These would all appear in `torch.jit`, so prefix everything but `overload` with a `_`